src/AutoRest.CSharp/Common/AutoRest/Plugins/Configuration.cs

```diff
@@ -596,5 +599,57 @@ private static bool ReadOption(JsonElement root, string option)
 
             return null;
         }
+
+        // Fetch CSharpProj configuration from Configuration
+        internal static CSharpProjConfiguration ToCSharpProjConfiguration() => new CSharpProjConfiguration
+        (
+            AbsoluteProjectFolder: AbsoluteProjectFolder,
+            AzureArm: AzureArm,
+            IsMgmtTestProject: MgmtTestConfiguration is not null,
+            LibraryName: LibraryName,
+            Namespace: Namespace,
+            SkipCSProjPackageReference: SkipCSProjPackageReference,
+            RelativeProjectFolder: RelativeProjectFolder,
+            Generation1ConvenienceClient: Generation1ConvenienceClient
+        );
+
+        // load CSharpProj configuration from Autorest input
+        internal static CSharpProjConfiguration InitializeCSharpProjConfig(IPluginCommunication autoRest, string defaultLibraryName, string defaultNamespace)
+        {
+            string outputFolder = GetOutputFolderOption(autoRest);
+            var projectFolder = GetProjectFolderOption(autoRest) ?? ProjectFolderDefault;
+            (var absoluteProjectFolder, var relativeProjectFolder) = ParseProjectFolders(outputFolder, projectFolder);
+
+            return new CSharpProjConfiguration(
+                AbsoluteProjectFolder: absoluteProjectFolder,
+                AzureArm: GetAzureArmOption(autoRest),
+                IsMgmtTestProject: MgmtTestConfiguration.HasMgmtTestConfiguration(autoRest),
+                LibraryName: GetLibraryNameOption(autoRest, defaultLibraryName),
+                Namespace: GetNamespaceOption(autoRest, defaultNamespace),
+                SkipCSProjPackageReference: GetSkipCSProjPackageReferenceOption(autoRest),
+                RelativeProjectFolder: relativeProjectFolder,
+                Generation1ConvenienceClient: GetGeneration1ConvenienceClientOption(autoRest)
+            );
+        }
```

Comment:
What is the difference between this method and the method `Initialize` above?

Reply:
In `StandaloneRunner`, we can directly convert ` Configuration` into `CSharpConfiguration`, instead of initializing twice.